xerosdk/apis/contacts.py

```diff
@@ -35,3 +36,15 @@ def post(self, data):
         """
 
         return self._post_request(data, Contacts.POST_CONTACTS)
+
+
+    def search_contact_by_display_name(self,display_name=None):
+        """
+        Search contact by display name
+        :param display_name: Xero Contact Name
+        :return: Contact
+        """
+        
+        response = self._get_request(Contacts.SEARCH_CONTACT.format(display_name))
+
+        return response['Contacts'][0] if len(response['Contacts']) > 0 else None
```

Comment:
can you paste whole payload for success and failure case here

Reply:
I am assuming by failure you mean if there is no contact with given name

Reply:
In case of success below is the response
`{'ContactID': '8b0ed9e2-240f-4248-a790-d8d5a6e315b7', 'ContactStatus': 'ACTIVE', 'Name': 'vaishnavi', 'FirstName': 'vaishnavi', 'LastName': '', 'EmailAddress': 'vaishnavi.mohan@fyle.in', 'BankAccountDetails': '', 'Addresses': [{'AddressType': 'STREET', 'City': '', 'Region': '', 'PostalCode': '', 'Country': ''}, {'AddressType': 'POBOX', 'City': '', 'Region': '', 'PostalCode': '', 'Country': ''}], 'Phones': [{'PhoneType': 'DDI', 'PhoneNumber': '', 'PhoneAreaCode': '', 'PhoneCountryCode': ''}, {'PhoneType': 'DEFAULT', 'PhoneNumber': '', 'PhoneAreaCode': '', 'PhoneCountryCode': ''}, {'PhoneType': 'FAX', 'PhoneNumber': '', 'PhoneAreaCode': '', 'PhoneCountryCode': ''}, {'PhoneType': 'MOBILE', 'PhoneNumber': '', 'PhoneAreaCode': '', 'PhoneCountryCode': ''}], 'UpdatedDateUTC': '/Date(1617790368040+0000)/', 'ContactGroups': [], 'IsSupplier': False, 'IsCustomer': False, 'ContactPersons': [], 'HasAttachments': False, 'HasValidationErrors': False}`

Reply:
In case of no contact found the response is:
`{'Id': 'c9430a07-0319-4193-a65a-4fe7eb2fa170', 'Status': 'OK', 'ProviderName': 'Fyle Staging', 'DateTimeUTC': '/Date(1617805239991)/', 'Contacts': []}`

Reply:
This should work `return response['Contacts'][0] if response['Contacts'] else None`